x-pack/plugins/apm/public/components/app/service_overview/service_overview_charts/use_filters_for_mobile_charts.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useMemo } from 'react';
+import { type QueryDslQueryContainer } from '@elastic/elasticsearch/lib/api/types';
+import { isNil, isEmpty } from 'lodash';
+import { ProcessorEvent } from '@kbn/observability-plugin/common';
+import { environmentQuery } from '../../../../../common/utils/environment_query';
+import { useApmParams } from '../../../../hooks/use_apm_params';
+import {
+  SERVICE_NAME,
+  TRANSACTION_TYPE,
+  PROCESSOR_EVENT,
+  HOST_OS_VERSION,
+  DEVICE_MODEL_IDENTIFIER,
+  NETWORK_CONNECTION_TYPE,
+  SERVICE_VERSION,
+} from '../../../../../common/elasticsearch_fieldnames';
+
+function termQuery<T extends string>(
+  field: T,
+  value: string | boolean | number | undefined | null
+): QueryDslQueryContainer[] {
+  if (isNil(value) || isEmpty(value)) {
+    return [];
+  }
+
+  return [{ term: { [field]: value } }];
+}
+
+export function useFiltersForMobileCharts() {
+  const {
+    path: { serviceName },
+    query: {
+      environment,
+      transactionType,
+      device,
+      osVersion,
+      appVersion,
+      netConnectionType,
+    },
+  } = useApmParams('/services/{serviceName}/overview');
```

Comment:
Don't you feel it would be better if this function receives as parameters all these values instead of using `useApmParams`? If we decide to use this on another page we'd have to add another path here and so on.

Reply:
I think it's more clear this way to just return the filters, but agree if we find out that it is being used in other pages and have to add more and more we can refactor.